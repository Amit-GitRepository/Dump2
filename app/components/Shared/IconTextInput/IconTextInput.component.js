import Icon from '../Icon/Icon.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './IconTextInput.styles';
import {colors} from '../../../themes/constants.styles';
import {TextInput, View} from 'react-native';

class IconTextInput extends Component {
  render () {
    const {accessibilityLabel, placeholder, iconName, iconSize, containerStyle, style, testID, type, ...extraProps} = this.props;
    const wrapperStyle = type === 'SECONDARY' ? [styles.container, styles.secondary, containerStyle] : [styles.container, containerStyle];
    return (
      <View style={wrapperStyle} accessibilityLabel={accessibilityLabel} testID={testID}>
        <Icon style={styles.icon} name={iconName} size={iconSize}/>
        <TextInput style={[styles.input, style]}
          placeholder={placeholder}
          underlineColorAndroid={colors.TRANSPARENT}
          placeholderTextColor={colors.PRIMARY_DISABLED_BG_TEXT} {...extraProps}/>
      </View>
    );
  }
}

IconTextInput.defaultProps = {
  containerStyle: {},
  style: {},
  placeholder: '',
  iconName: null,
  iconSize: 35
};

IconTextInput.propTypes = {
  accessibilityLabel: PropTypes.string,
  type: PropTypes.oneOf(['SECONDARY', 'PRIMARY']),
  containerStyle: PropTypes.object,
  style: PropTypes.object,
  placeholder: PropTypes.string,
  iconName: PropTypes.string,
  iconSize: PropTypes.number,
  testID: PropTypes.string
};

export default IconTextInput;
