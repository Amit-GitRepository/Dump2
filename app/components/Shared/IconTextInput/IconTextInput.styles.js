import {colors, fonts} from '../../../themes/constants.styles';

export default {
  container: {
    alignSelf: 'stretch',
    borderWidth: 2,
    borderRadius: 4,
    borderColor: colors.LANDING_TITLE_BG,
    flexDirection: 'row',
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST
  },
  secondary: {
    borderColor: colors.PRIMARY_DISABLED_BG_TEXT,
    borderWidth: 1
  },
  icon: {
    alignSelf: 'center',
    color: colors.PRIMARY_DISABLED_BG_TEXT,
    padding: 5,
    backgroundColor: colors.TRANSPARENT
  },
  input: {
    flex: 1,
    alignSelf: 'stretch',
    paddingVertical: 10,
    paddingHorizontal: 0,
    borderRadius: 4,
    fontSize: fonts.FONT_SIZE_NORMAL
  }
};
