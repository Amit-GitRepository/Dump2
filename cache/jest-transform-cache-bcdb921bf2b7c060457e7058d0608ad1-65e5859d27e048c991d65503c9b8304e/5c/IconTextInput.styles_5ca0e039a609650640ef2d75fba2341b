Object.defineProperty(exports,"__esModule",{value:true});var cov_2bndyw55r1=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/IconTextInput/IconTextInput.styles.js',hash='ba68b9aabac3bd712dc755626eff412ddcf7ba2a',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/IconTextInput/IconTextInput.styles.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
container:{
alignSelf:'stretch',
borderWidth:2,
borderRadius:4,
borderColor:_constants.colors.LANDING_TITLE_BG,
flexDirection:'row',
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST},

icon:{
alignSelf:'center',
color:_constants.colors.PRIMARY_DISABLED_BG_TEXT,
padding:5,
backgroundColor:_constants.colors.TRANSPARENT},

input:{
flex:1,
alignSelf:'stretch',
paddingVertical:10,
paddingHorizontal:0,
borderRadius:4,
fontSize:_constants.fonts.FONT_SIZE_NORMAL}};