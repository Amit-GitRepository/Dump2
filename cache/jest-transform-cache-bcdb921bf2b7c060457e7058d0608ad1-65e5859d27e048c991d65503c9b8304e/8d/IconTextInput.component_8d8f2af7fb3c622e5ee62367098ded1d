Object.defineProperty(exports,"__esModule",{value:true});var cov_223ccszm9e=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/IconTextInput/IconTextInput.component.js',hash='1c2c986d85e4a6d0b9ead8546578062699561c9a',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/IconTextInput/IconTextInput.component.js',statementMap:{'0':{start:{line:10,column:112},end:{line:10,column:122}},'1':{start:{line:11,column:4},end:{line:19,column:6}},'2':{start:{line:23,column:0},end:{line:29,column:2}},'3':{start:{line:31,column:0},end:{line:39,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:9,column:2},end:{line:9,column:3}},loc:{start:{line:9,column:12},end:{line:20,column:3}},line:9}},branchMap:{},s:{'0':0,'1':0,'2':0,'3':0},f:{'0':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Icon=require('../Icon/Icon.component');var _Icon2=_interopRequireDefault(_Icon);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _IconTextInput=require('./IconTextInput.styles');var _IconTextInput2=_interopRequireDefault(_IconTextInput);
var _constants=require('../../../themes/constants.styles');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _objectWithoutProperties(obj,keys){var target={};for(var i in obj){if(keys.indexOf(i)>=0)continue;if(!Object.prototype.hasOwnProperty.call(obj,i))continue;target[i]=obj[i];}return target;}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

IconTextInput=function(_Component){_inherits(IconTextInput,_Component);function IconTextInput(){_classCallCheck(this,IconTextInput);return _possibleConstructorReturn(this,(IconTextInput.__proto__||Object.getPrototypeOf(IconTextInput)).apply(this,arguments));}_createClass(IconTextInput,[{key:'render',value:function render()
{cov_223ccszm9e.f[0]++;var _ref=(cov_223ccszm9e.s[0]++,
this.props),accessibilityLabel=_ref.accessibilityLabel,placeholder=_ref.placeholder,iconName=_ref.iconName,iconSize=_ref.iconSize,containerStyle=_ref.containerStyle,style=_ref.style,testID=_ref.testID,extraProps=_objectWithoutProperties(_ref,['accessibilityLabel','placeholder','iconName','iconSize','containerStyle','style','testID']);cov_223ccszm9e.s[1]++;
return(
_react2.default.createElement(_reactNative.View,{style:[_IconTextInput2.default.container,containerStyle],accessibilityLabel:accessibilityLabel,testID:testID},
_react2.default.createElement(_Icon2.default,{style:_IconTextInput2.default.icon,name:iconName,size:iconSize}),
_react2.default.createElement(_reactNative.TextInput,_extends({style:[_IconTextInput2.default.input,style],
placeholder:placeholder,
underlineColorAndroid:_constants.colors.TRANSPARENT,
placeholderTextColor:_constants.colors.PRIMARY_DISABLED_BG_TEXT},extraProps))));


}}]);return IconTextInput;}(_react.Component);cov_223ccszm9e.s[2]++;


IconTextInput.defaultProps={
containerStyle:{},
style:{},
placeholder:'',
iconName:null,
iconSize:35};cov_223ccszm9e.s[3]++;


IconTextInput.propTypes={
accessibilityLabel:_propTypes2.default.string,
containerStyle:_propTypes2.default.object,
style:_propTypes2.default.object,
placeholder:_propTypes2.default.string,
iconName:_propTypes2.default.string,
iconSize:_propTypes2.default.number,
testID:_propTypes2.default.string};exports.default=


IconTextInput;