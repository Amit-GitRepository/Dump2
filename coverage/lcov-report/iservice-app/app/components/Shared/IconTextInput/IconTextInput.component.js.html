<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for iservice-app/app/components/Shared/IconTextInput/IconTextInput.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">iservice-app/app/components/Shared/IconTextInput</a> IconTextInput.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">126x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">126x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import Icon from '../Icon/Icon.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './IconTextInput.styles';
import {colors} from '../../../themes/constants.styles';
import {TextInput, View} from 'react-native';
&nbsp;
class IconTextInput extends Component {
  render () {
    const {accessibilityLabel, placeholder, iconName, iconSize, containerStyle, style, testID, type, ...extraProps} = this.props;
    const wrapperStyle = type === 'SECONDARY' ? [styles.container, styles.secondary, containerStyle] : [styles.container, containerStyle];
    return (
      &lt;View style={wrapperStyle} accessibilityLabel={accessibilityLabel} testID={testID}&gt;
        &lt;Icon style={styles.icon} name={iconName} size={iconSize}/&gt;
        &lt;TextInput style={[styles.input, style]}
          placeholder={placeholder}
          underlineColorAndroid={colors.TRANSPARENT}
          placeholderTextColor={colors.PRIMARY_DISABLED_BG_TEXT} {...extraProps}/&gt;
      &lt;/View&gt;
    );
  }
}
&nbsp;
IconTextInput.defaultProps = {
  containerStyle: {},
  style: {},
  placeholder: '',
  iconName: null,
  iconSize: 35
};
&nbsp;
IconTextInput.propTypes = {
  accessibilityLabel: PropTypes.string,
  type: PropTypes.oneOf(['SECONDARY', 'PRIMARY']),
  containerStyle: PropTypes.object,
  style: PropTypes.object,
  placeholder: PropTypes.string,
  iconName: PropTypes.string,
  iconSize: PropTypes.number,
  testID: PropTypes.string
};
&nbsp;
export default IconTextInput;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 16:14:06 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
