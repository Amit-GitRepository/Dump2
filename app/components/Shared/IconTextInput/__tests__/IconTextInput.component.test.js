import IconTextInput from '../IconTextInput.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer'; // Enzyme adapter for React 16
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('IconTextInput component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<IconTextInput/>);
    const component = renderer.create(<IconTextInput />).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });
  it('renders correctly with icon', () => {
    const wrapper = shallow(<IconTextInput iconName='trueid'/>);
    expect(wrapper).toBeDefined();
  });
});
