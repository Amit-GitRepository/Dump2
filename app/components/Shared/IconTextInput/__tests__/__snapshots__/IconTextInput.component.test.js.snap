// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconTextInput component renders correctly 1`] = `
<View
  accessibilityLabel={undefined}
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#605E71",
        "borderRadius": 4,
        "borderWidth": 2,
        "flexDirection": "row",
      },
      Object {},
    ]
  }
  testID={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 35,
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "color": "#B9C1C9",
          "padding": 5,
        },
        Object {
          "fontFamily": "iservice",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <TextInput
    allowFontScaling={true}
    placeholder=""
    placeholderTextColor="#B9C1C9"
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "borderRadius": 4,
          "flex": 1,
          "fontSize": 16,
          "paddingHorizontal": 0,
          "paddingVertical": 10,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
